cumulusci/tasks/apex/custom_settings.py

```diff
@@ -0,0 +1,96 @@
+""" a task for waiting on a specific custom settings value """
+
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+from cumulusci.core.utils import process_bool_arg
+
+
+class CustomSettingValueWait(BaseSalesforceApiTask):
+    """ CustomSettingValueWait polls an org until the specific value exists in a custom settings field """
+
+    name = "CustomSettingValueWait"
+
+    task_options = {
+        "object": {
+            "description": "Name of the SObject to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "field": {
+            "description": "Name of the field on the Sobject to query. Can include the %%%NAMESPACE%%% token. ",
+            "required": True,
+        },
+        "value": {
+            "description": "Value of the field to wait for (String or Boolean). ",
+            "required": True,
+        },
+        "managed": {
+            "description": (
+                "If True, will inject the project's namespace prefix to replace %%%NAMESPACE%%% tokens in the object or field "
+                "Defaults to False or no namespace."
+            ),
+            "required": False,
+        },
+        "poll_interval": {
+            "description": (
+                "Seconds to wait before polling for batch job completion. "
+                "Defaults to 10 seconds."
+            ),
+        },
+    }
+
+    def _run_task(self):
+        self.poll_interval_s = int(self.options.get("poll_interval", 10))
+
+        self.object_name = self.options["object"]
+        self.field_name = self.options["field"]
+        self.check_value = self.options["value"]
+
+        # Process namespace tokens
+        managed = self.options.get("managed") or False
+        namespace = self.project_config.project__package__namespace
+        namespace_prefix = ""
+        if managed:
+            namespace_prefix = namespace + "__"
+
+        self.object_name = self.object_name.replace("%%%NAMESPACE%%%", namespace_prefix)
+        self.field_name = self.field_name.replace("%%%NAMESPACE%%%", namespace_prefix)
+
+        self._poll()  # will block until poll_complete
+
+        self.logger.info("Value Matched.")
+
+        return True
+
+    def _poll_action(self):
+        query_results = self.sf.query(self._object_query)
+
+        self.record = query_results["records"][0]
+
+        self.poll_complete = not self._poll_again()
+
+    def _poll_again(self):
+        return not self.success
+
+    @property
+    def success(self):
+        self.field_value = self.record[self.field_name]
+
+        if isinstance(self.field_value, (bool)):
+            self.check_value = process_bool_arg(self.check_value)
+            self.field_value = process_bool_arg(self.field_value)
+        elif isinstance(self.field_value, (int, float)):
+            self.check_value = float(self.check_value)
```

Comment:
Just out of curiosity, do you know of any use-cases for floats? Comparing floats always makes me slightly nervous so I'd like to know what range they would be in.

Reply:
In testing, any numeric field in Salesforce is retrieved with a number to right decimal point, even if that's zero and even if the Number field in Salesforce is configured with a scale of 0. This is consistent with how all Number fields are handled in Salesforce as well.

Reply:
Comparing floats always sets off a clarion in my head, but I don't know that it will cause a problem in this situation so I think we should go ahead with it.